miden-lib/build.rs

```diff
@@ -164,6 +172,61 @@ fn decrease_pow(line: io::Result<String>) -> io::Result<String> {
     Ok(line)
 }
 
+/// Generates `kernel_procs_v0.bin` and `kernel_proc_offsets.masm` files based on the kernel library
+fn generate_procedures_metadata(kernel: KernelLibrary) -> Result<()> {
+    let (_, module_info, _) = kernel.into_parts();
+
+    let to_exclude = BTreeSet::from_iter(["exec_kernel_proc"]);
+    let mut proc_digests_as_felts = vec![];
+    let mut generated_consts = String::new();
+    let mut generated_accessors = String::new();
+
+    module_info
+        .procedures()
+        .filter(|(_, proc_info)| !to_exclude.contains::<str>(proc_info.name.as_ref()))
+        .enumerate()
+        .for_each(|(offset, (_, proc_info))| {
```

Comment:
One tricky thing here is that the order of procedures should never change, even if changes in the kernel library. We may need to create a separate array where we list procedure names in the relevant order and use that array to figure out indexes for each procedure.

Reply:
Is it really needed? I solve this by regeneration of offsets file. But not sure if users have access to that constants. Nevertheless, all functions are actually in alphabetic order, but adding of new function will force offsets to change.

Reply:
If we need to freeze ordering, I can offer not to regenerate offsets file, but use it as source of offsets instead.

Reply:
The constants would be managed by us, but it is important to make sure the ordering of procedures doesn't change even if we rename them or add new procedures later on. Basically, this list of constants would be kind of like "append-only".

Reply:
> If we need to freeze ordering, I can offer not to regenerate offsets file, but use it as source of offsets instead.

Do you mean the MASM file?

Reply:
> > If we need to freeze ordering, I can offer not to regenerate offsets file, but use it as source of offsets instead.
> 
> Do you mean the MASM file?

Yes!

Reply:
That could work. Let's try it!